### PR TITLE
[tesla] Add support for setting charging amps and sentry mode

### PR DESCRIPTION
**Pull request by kaikreuzer**
Implements https://github.com/openhab/openhab-addons/issues/11483

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

**Comment by ipweb2000** [comment]
In the documentation, I guess it should read "Number:ElectricCurrent" and not "ElectricPotential" for chargingAmps
| chargingamps              | Number:ElectricPotential | Set Charging Amps             | Current (Ampere) to use for charging

**Comment by kaikreuzer** [comment]
Thanks @ipweb2000, well spotted. I corrected it.

File: bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaVehicleHandler.java [hunk 1]
```diff
@@ -263,6 +265,18 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                             }
                             break;
                         }
+                        case CHARGE_AMPS:
+                            if (command instanceof QuantityType) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaVehicleHandler.java:269]: You could also accept Decimal type?

Comment (kaikreuzer) [reply bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaVehicleHandler.java]: I guess so, yes. I've changed the code accordingly.

File: bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaVehicleHandler.java [hunk 2]
```diff
@@ -266,9 +266,14 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                             break;
                         }
                         case CHARGE_AMPS:
-                            if (command instanceof QuantityType) {
-                                @SuppressWarnings({ "unchecked", "null" })
-                                int amps = ((QuantityType<ElectricCurrent>) command).toUnit(Units.AMPERE).intValue();
+                            if (command instanceof DecimalType) {
+                                int amps = ((DecimalType) command).intValue();
+                                if (command instanceof QuantityType<?>) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaVehicleHandler.java:271]: Sorry but I believe it is not good. There is no hierarchy between DecimalType and QuanyityType. 
You have to separate the two cases.

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaVehicleHandler.java (hunk 2 above)
Comment (kaikreuzer) [reply bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaVehicleHandler.java]: Thanks for looking so closely! You're clearly right, I have fixed this. Hope this time I got it correct!

**Review by lolodomo — APPROVED** [review]

**Pull request merged**